lib/libzfs/libzfs_sendrecv.c

```diff
@@ -3801,8 +3801,9 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 	}
 
 	ASSERT(strstr(drrb->drr_toname, sendfs) == drrb->drr_toname);
-	ASSERT(chopprefix > drrb->drr_toname);
-	ASSERT(chopprefix <= drrb->drr_toname + strlen(drrb->drr_toname));
+	ASSERT(chopprefix > drrb->drr_toname || strrchr(sendfs, '/') == NULL);
+	ASSERT(chopprefix <= drrb->drr_toname + strlen(drrb->drr_toname) ||
+	    strrchr(sendfs, '/') == NULL);
```

Comment:
might as well change this to just `strchr()`

Reply:
Could you please clarify this? Are you suggesting to remove `chopprefix <= drrb->drr_toname + strlen(drrb->drr_toname)` from the second `ASSERT()`?

Reply:
No, I'm just saying we could use `strchr()` instead of `strrchr()` here. Minor nit really.

Reply:
Ah, sorry, i had missed the extra "r" in "strrchr()". I'm going to force-push in a bit.